src/main/java/com/synopsys/integration/blackduck/nexus3/task/scan/RepositoryScanner.java

```diff
@@ -210,21 +210,24 @@ private boolean shouldScan(TaskStatus status) {
                 assetWrapper.addPendingToBlackDuckPanel("Scan uploaded to Black Duck, waiting for update.");
             }
         } catch (BlackDuckApiException e) {
-            logger.error(BLACK_DUCK_COMMUNICATION_FORMAT, e.getMessage());
-            logger.debug(e.getMessage(), e);
-            assetWrapper.removeAllBlackDuckData();
-            assetWrapper.addFailureToBlackDuckPanel(e.getMessage());
-        } catch (IntegrationException e) {
-            logger.error("Error scanning asset: {}. Reason: {}", name, e.getMessage());
-            logger.debug(e.getMessage(), e);
-            assetWrapper.removeAllBlackDuckData();
-            assetWrapper.addFailureToBlackDuckPanel(e.getMessage());
+            String errorMessage = String.format(BLACK_DUCK_COMMUNICATION_FORMAT, e.getMessage());
+            handleScanException(assetWrapper, errorMessage, e);
+        } catch (IntegrationException | IllegalArgumentException e) {
+            String errorMessage = String.format("Error scanning asset: %s, version: %s, path: %s. Reason: %s", projectName, version, fullPath, e.getMessage());
+            handleScanException(assetWrapper, errorMessage, e);
```

Comment:
Adding the catch for `IllegalArgumentException` and improving the log message on line 190 because a customer is experiencing and issue where the `ScanBatchBuilder` is throwing an `IllegalArgumentException` for the project name or version missing.